### PR TITLE
Fix CMake install issue and HAVE_CPU_SET_T issue

### PR DESCRIPTION
**Pull request by mahrud**
I'm working on using the trunk branch in Macaulay2. This PR fixes issues that I run into while working on that.

Is `HAVE_CPU_SET_T` expected to be defined on a typical Linux build?

cc: @mikestillman

### CODE REVIEW

**Comment by wbhart** [comment]
I believe so. I think @tthsqe12 can say more about it, though he is temporarily without internet access so it may be a short while before he replies.

**Comment by isuruf** [comment]
The line https://github.com/wbhart/flint2/blob/2b8f8acb317c265db99f828e7baf3266f07f92a7/config.h.in#L291

should be `#cmakedefine01` instead of `#cmakedefine`

**Comment by mahrud** [comment]
Ohh I did not know `#cmakedefine01` was a thing!! I'll fix this in a moment. Thanks!

File: thread_pool.h [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
-#if HAVE_CPU_SET_T
+#ifdef HAVE_CPU_SET_T
```
Comment (isuruf) [inline thread_pool.h]: You don't need this change anymore

Comment (mahrud) [reply thread_pool.h]: I made a mistake with replacing the commit. It should be fixed now.

I was wondering though, is there a reason why `#cmakedefine01` isn't used for anything else?

**Comment by tthsqe12** [comment]
A typical linux build should have cpu_set_t, for it is the bitset used by the kernel for sched_setaffinity. This is only used for pinning threads to cores, which is not vital functionality (unless you are try to benchmarket).

**Comment by wbhart** [comment]
This seems harmless enough, so I'm merging it now.

**Pull request merged**